src/Box.hpp

```diff
@@ -17,7 +17,7 @@ class Box
     Box(sf::IntRect boxRect);
 
     // Destructor
-    ~Box() {};
+    virtual ~Box() {};
```

Comment:
So this is overriden in children?

Reply:
Not really. Destructors are special (just like constructors). The subclass will have its destructor called first, then the base class destructor will be called. So both will still be called. The reason it is important to have a virtual destructor in the base class is very well described [in this SO post](https://stackoverflow.com/questions/461203/when-to-use-virtual-destructors).